File: day1/main.py
import os

def count(step):
    levels = [int(x) for x in open(os.path.join(os.path.dirname(__file__), "input.txt")).read().split("\n")]
    sum = 0
    for i in range(len(levels) - step):
        if levels[i] < levels[i + step]:
            sum += 1
    return sum

print(count(1), count(3))

File: day1/main.lua
local levels = {}
for line in io.lines("input.txt") do
    table.insert(levels, tonumber(line))
end

local function count(step)
    local sum = 0
    for i = 1, #levels - step do
        if levels[i] < levels[i + step] then
            sum = sum + 1
        end
    end
    return sum
end

print(count(1), count(3))
